src/core/unit.c

```diff
@@ -3380,7 +3380,10 @@ int unit_write_drop_in(Unit *u, UnitSetPropertiesMode mode, const char *name, co
         if (!dir)
                 return -EINVAL;
 
-        r = write_drop_in(dir, u->id, 50, name, data);
+        prefixed = strjoina("# This is a drop-in unit file extension, created via \"systemctl set-property\" or an equivalent operation. Do not edit.\n",
```

Comment:
Why should this not be edited? If the user changes this, systemd will read it again and systemctl set-property should still work again?